integration-tests/contract-execute-on-chain-view.spec.ts

```diff
@@ -1,19 +1,110 @@
 import { codeViewsTopLevel } from "../packages/taquito-local-forging/test/data/contract_views_top_level";
 import { CONFIGS } from "./config";
 import BigNumber from 'bignumber.js';
-import { Protocols } from "@taquito/taquito";
+import { Protocols, ViewSimulationError } from "@taquito/taquito";
 import { HttpResponseError } from "@taquito/http-utils";
 
-CONFIGS().forEach(({ lib, rpc, setup }) => {
+CONFIGS().forEach(({ lib, protocol, rpc, setup }) => {
   const Tezos = lib;
- 
+  const jakartanet = protocol === Protocols.PtJakart2 ? test: test.skip;
+  const kathmandunet = protocol === Protocols.PtKathman ? test: test.skip;
+
   describe(`On chain views using the contract API: ${rpc}`, () => {
 
     beforeEach(async (done) => {
       await setup(true)
       done()
     })
-    test(`As a user I want to originate a smart contract having top level views and simulate the views execution`, async (done) => {
+    kathmandunet(`On K protocol as a user I want to originate a smart contract having top level views and simulate the views execution`, async (done) => {
```

Comment:
We can have just 1 test, and it should work on jakartanet and Kathmandu as the distinction is handle in the code